### ISSUE TITLE
Provide better support for mac_os_x

### ISSUE THREAD

**Issue by Brantone** [gildegoma/chef-android-sdk#25]
`expect` is typically default installed on OSX, so not always needed, thus 'homebrew' not required either (and some instances 'homebrew' might not be allowed anyway).

More importantly `/etc/profile.d` does not exist, thus https://github.com/gildegoma/chef-android-sdk/blob/master/recipes/default.rb#L91 fails - although `set_environment_variables` can be used to to circumvent, would be nice still set and have legit.

**Comment by gildegoma** [comment]
@Brantone thanks for filing these issues. At the moment, this cookbook is [not *officially* supporting the Mac OS X platform](https://github.com/gildegoma/chef-android-sdk/blob/cb67ebd25d656f47fb581ed9a81c9d11c50abb86/metadata.rb#L15-L16).

Instead of patching isolated items (e.g. `expect` optional setup - #26), I'd prefer to address `mac_os_x` support as a whole. Relying on homebrew's [`android-sdk`](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/android-sdk.rb) could be a good start, although Hombrew default approach is to only provide the latest stable release (i.e. no easy version selection). Or do you think that this cookbook should be indeed offer an "advanced" alternative to the hombrew way?

**Comment by Brantone** [comment]
Ultimately it would be nice to see this move towards LWRP, but in interim, the 2 main hurdles to some semblance of osx support is `expect` and `/etc/profile.d`, both of which being pretty straight forward to solve. Because `expect` is usually standard on OSX machines, could take a gamble on it being installed ... as for `profile.d`, the OSX equivalent is `/etc/paths.d` and is a modified version of android-sdk.sh.erb

**Comment by gildegoma** [comment]
@Brantone and then this cookbook will be fully independent of any Mac package manager (e.g. no call to any `package` resource), correct?

**Comment by Brantone** [comment]
That's my current line of reasoning, yes ... always open to suggestions on something I might have missed.

**Comment by gildegoma** [comment]
So far LGTM, are you willing to provide a pull request for this then?

**Comment by Brantone** [comment]
You mean LWRP (light weight resource provider)? ... I wish I had time to do it sooner rather than later :( might not be for a while yet.

**Comment by gildegoma** [comment]
No, I am currently not interested to introduce the LWRP. Only the points you mentioned above:

* use `package 'expect'` only on a linux platform (or not on a mac platform) - replacing #26. We then have no dependency on `homebrew`.
*  deal with `/etc/paths.d` when running on a mac platform
